stock_lot_analytic/tests/test_stock_lot_analytic.py

```diff
@@ -1,29 +1,58 @@
 # Copyright 2023 Quartile Limited
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
 
-from odoo.addons.purchase_stock.tests.test_create_picking import TestCreatePicking
+from odoo.fields import Command
+from odoo.tests.common import TransactionCase
 
 
-class TestStockLotAnalytic(TestCreatePicking):
-    def test_stock_lot_analytic_with_incoming_picking(self):
-        self.product_id_1.tracking = "lot"
-        self.po = self.env["purchase.order"].create(self.po_vals)
-        self.po.order_line.write(
+class TestStockLotAnalytic(TransactionCase):
+    @classmethod
+    def setUpClass(cls):
+        super().setUpClass()
+        cls.product = cls.env["product.product"].create(
+            {
+                "name": "Test Product",
+                "type": "product",
+                "tracking": "lot",
+            }
+        )
+        cls.analytic_account = cls.env.ref("analytic.analytic_agrolait")
+        cls.vendor = cls.env["res.partner"].create({"name": "Test Vendor"})
+        cls.po = cls.env["purchase.order"].create(
             {
-                "analytic_distribution": {
-                    str(self.env.ref("analytic.analytic_agrolait").id): 100.0
-                }
+                "partner_id": cls.vendor.id,
+                "order_line": [
+                    Command.create(
+                        {
+                            "product_id": cls.product.id,
+                            "product_qty": 2.0,
+                            "product_uom": cls.product.uom_id.id,
+                            "price_unit": 10.0,
+                            "analytic_distribution": {
+                                str(cls.analytic_account.id): 100.0
+                            },
+                        }
+                    )
+                ],
             }
         )
+
+    def test_stock_lot_analytic_with_incoming_picking(self):
         self.po.button_confirm()
-        self.picking = self.po.picking_ids[0]
-        for ml in self.picking.move_line_ids:
+        picking = self.po.picking_ids.filtered(
+            lambda p: p.picking_type_id.code == "incoming"
+        )
```

Comment:
I think there is no point in doing `filtered()`.
```suggestion
        picking = self.po.picking_ids
```